packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection, Org, SfProject } from '@salesforce/core';
+import {
+  SourceTracking,
+  SourceTrackingOptions
+} from '@salesforce/source-tracking';
+
+export class SourceTrackingService {
+  public static async createSourceTracking(
+    projectPath: string,
+    connection: Connection
+  ): Promise<SourceTracking> {
+    // Since SourceTracking is initialized with an SfProject, which
+    // contains the project path, and PR #4643 made it so that VSCE is
+    // running with process.cwd set as the project root, there
+    // is no need to call process.chdir here as has been done in VSCE
+    // with other core types like Config and ConfigAggregator.
+    const project = await SfProject.resolve(projectPath);
+    const org = await Org.create({ connection });
+    const options: SourceTrackingOptions = {
+      org,
+      project,
+      ignoreLocalCache: true,
+      subscribeSDREvents: true,
+      ignoreConflicts: false
+    };
+    const sourceTracking = await SourceTracking.create(options);
+    await sourceTracking.ensureLocalTracking();
```

Comment:
With ignoreLocalCache: false, newly created files would not be seen by STL and it would allow a newly created Apex file to overwrite more recently made changes on the server.  

By setting ignoreLocalCache to true, and calling ensureLocalTracking(), we force the STL instance to re-load its cache, which causes it to re-read the filesystem.  When it does that it "sees" the newly created files and will throw when conflicts exist in this scenario.

What is needed:
<img width="500" alt="2023-03-16_23-01-14" src="https://user-images.githubusercontent.com/46458081/225977627-feba4608-e375-4b17-bb94-ad3268aaff1f.png">

How to get there:
<img width="500" alt="2023-03-16_23-00-03" src="https://user-images.githubusercontent.com/46458081/225977719-7d0be489-b359-405b-99ca-d4ac7fc6c541.png">

The local repo is private to the STL instance, otherwise we could make a direct call to it to bust the cache.  As it is currently, ignoreLocalCache=true and ensureLocalTracking() is the only way to get there:
<img width="500" alt="2023-03-16_23-00-15" src="https://user-images.githubusercontent.com/46458081/225978003-b42b6d20-9f7c-493e-bc5e-d28ada94310e.png">